data-sources/api/src/commonMain/kotlin/EpisodeSort.kt

```diff
@@ -110,22 +131,35 @@ sealed class EpisodeSort : Comparable<EpisodeSort> {
     }
 }
 
+private fun getSpecialByRaw(raw: String): EpisodeSort {
+    val type = EpisodeType.entries.firstOrNull { entry -> raw.startsWith(entry.value, ignoreCase = true) }
+    if (type == null || raw.length < type.value.length) return EpisodeSort.Unknown(raw)
+    val numStr = raw.substringAfter(type.value);
+    return Special(type, numStr.toFloatOrNull())
```

Comment:
```suggestion
    return Special(type, numStr.toFloatOrNull())
```

这个逻辑可能是有问题的:
- `SP-1` 期望是 `Unknown("SP-1")` 而不是 `Special(SP, null)`. 目前 `EpisodeSort(-1)` 也是会返回 Unknown 的, 行为需要保持一致. 允许负数也会导致 compare 出问题, 目前应该是拿字符串比较的
- `SPAB` 期望是 `Unknown("SPAB")` 而不是 `Special(SP, null)`